### PR TITLE
Piers fix js week 5

### PR DESCRIPTION
**Pull request by piers-williams**

### CODE REVIEW

File: web-training/JavaScript/Week 5 - Working with Arrays/Ex1/Ex1.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <html>
 	<head>
-		<title>Ex1 Yours</title>
+	<title>Ex1 Target</title>
```
Comment (Alex-Kondratov-Redgate) [inline web-training/JavaScript/Week 5 - Working with Arrays/Ex1/Ex1.html:3]:
```suggestion
	<title>Ex1 Yours</title>
```

File: web-training/JavaScript/Week 5 - Working with Arrays/Ex1/Ex1.html [hunk 2]
```diff
@@ -32,10 +32,61 @@
...
+			const table = document.getElementById("student-table");
+			const tbody = table.getElementsByTagName("tbody")[0];
+			tbody.innerHTML = "";
+			
+			const selectValue = document.getElementById("sort").value;
+			const classValue = document.getElementById("class").value;
+			
+			const elements = x
+			.map((x, index) => {return {...x, index: index};})
+			.map(person => {
```
Comment (Alex-Kondratov-Redgate) [inline web-training/JavaScript/Week 5 - Working with Arrays/Ex1/Ex1.html:49]: Perhaps we can extract a function that generates a single row and then ask people to use it in `map`?

File: web-training/JavaScript/Week 5 - Working with Arrays/Ex1/Ex1_target.html [hunk 3]
```diff
@@ -45,21 +45,46 @@
...
+			const selected = document.querySelector('input[name="selection"]:checked').value;
+			x.splice(selected, 1, {Name: document.getElementById("new-name").value, Score: document.getElementById("new-score").value});
+			loadTable();			
+		}
+		
+		function report(){
+			const reportDiv = document.getElementById("report-location");
+			var average = document.createElement("p");
+			const averageScore = x.map(a => a.Score).reduce((a, c) => a + c) / x.length;
+			average.innerHTML = `Average score: ${averageScore}`;
```
Comment (samilamti) [inline web-training/JavaScript/Week 5 - Working with Arrays/Ex1/Ex1_target.html:81]:
```suggestion
			average.textContent = `Average score: ${averageScore}`;
```

File: web-training/JavaScript/Week 5 - Working with Arrays/Ex1/Ex1_target.html [hunk 4]
```diff
@@ -45,21 +45,46 @@
...
+			var csv = document.createElement("pre");
+			csv.innerHTML = "Name, Score\n" + x.map(person => `${person.Name}, ${person.Score}`).join('\n');
```
Comment (samilamti) [inline web-training/JavaScript/Week 5 - Working with Arrays/Ex1/Ex1_target.html:83]:
```suggestion
			csv.textContent = "Name, Score\n" + x.map(person => `${person.Name}, ${person.Score}`).join('\n');
```

File: web-training/JavaScript/Week 5 - Working with Arrays/Ex1/Ex1.html [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <html>
 	<head>
-		<title>Ex1 Yours</title>
+	<title>Ex1 Yours</title>
 		<script>
 		const x = [
```
Comment (samilamti) [inline web-training/JavaScript/Week 5 - Working with Arrays/Ex1/Ex1.html:5]:
```suggestion
		const students = [
```

**Review by samilamti — CHANGES_REQUESTED** [review]
Let's teach them good habits by injecting text with `textContent`. Let's also rename `x` to `students`; I noticed some attendees struggle to remember what `x` was.

**Pull request merged**